src/lib.rs

```diff
@@ -68,7 +71,7 @@ pub use host::{Host, ffi as host_ffi};
 pub use host::data::open as data_open;
 pub use mustache::{MapBuilder, VecBuilder};
 pub use package::{Package, ffi as package_ffi};
-pub use package::providers::Providers;
+pub use package::providers::{Providers, ProviderFactory};
```

Comment:
Good catch! The Agent is the only project using that, and probably shouldn't in future. Best to export it for now.